megatron/core/datasets/blended_megatron_dataset_builder.py

```diff
@@ -232,52 +232,61 @@ def _build_blended_dataset_splits(self,) -> List[Optional[TopLevelDataset]]:
                 blend = self.config.blend_per_split[i]
                 if blend is not None:
                     prefixes, weights = blend
-                    if weights is not None:
-                        weights = normalize(weights)
-
-                    # Blend consists of a sigle prefix
-                    if len(prefixes) == 1:
-                        blended_datasets[i] = self._build_megatron_dataset_splits(
-                            prefixes[0], split_spoof, sizes_spoof
-                        )[i]
-                        continue
-
-                    # Build mid-level datasets
-                    if weights is None:
-                        sizes_per_dataset = [[None for split in Split] for prefix in prefixes]
+                    if self.config.multiple_valid_sets and not isinstance(weights[0], (int, float)):
```

Comment:
So if we using `self.config.multiple_valid_sets`, the `weights` keep the name of the dataset? 
Then we should probably at least give a warning if the flag is true, but the weights are numbers

Reply:
I changed it a bit so that we don't have `weights` keeping dataset names